src/main/java/io/strimzi/kafka/bridge/tracing/OpenTelemetryHandle.java

```diff
@@ -65,6 +65,7 @@ public String serviceName(BridgeConfig config) {
     @Override
     public void initialize() {
         System.setProperty("otel.metrics.exporter", "none"); // disable metrics
+        System.setProperty("otel.exporter.otlp.protocol", "http/protobuf"); // force the use of http/protobuf rather than GRPC this will (silently) override the OTEL_EXPORTER_OTLP_PROTOCOL environment variable
```

Comment:
What is the impact on existing users?

Reply:
So currently users have to use a tracing backend exposing the OTLP related port with gRPC support (i.e. Jaeger backend has port 4317). Staying this way would not work, so it's a kind of braking change. The user has to expose the OTLP related port with HTTP support (i.e. Jaeger backend has port 4318).

Other question I have here is why forcing protobuf if the user would like to use JSON for example? The HTTP port also supports `http/json` as OTLP protocol, or? So if user doesn't set it, it would default to `grpc` which would be not supported by us, maybe we should "force" the user to set it, raising a warning/exception if not set? Or we could default to `http/protobuf` if not set.

Reply:
So I guess we should not do it this way? We should certainly not hardcode it into the code. But we probably first deprecate what we have and give users some options to migrate? It seems to be like this would essentially break at the time of upgrade.

Reply:
To avoid having a Kotlin dependency through okhttp we need to replace it with the JDK http client which doesn't support gRPC so I don't see a way for deprecation in this scenario. The OpenTelemetry library can use okhttp or the JDK http client but not both at same time to allow users to easily migrate.
The other way would be taking okhttp for one more release but saying to move using http/protobuf as protocol instead grpc. But this solution would not allow us to get rid of Kotlin dependency.

Reply:
@scholzj just to make my point clearer ... I see your point and I agree that a smooth migration with deprecation is better. To go this way we should first verify that okhttp really works when setting the OTLP protocol as `http/protobuf` instead of the default `grpc`. So if we go this way, we should hold on with this change and taking it for a future release (not the next one).

Reply:
Yeah, I guess if OkHttp supports both, we should stick with it for a few more releases and just instruct users to switch to the HTTP protocol as GRPC will not be supported in the future.

Reply:
>  To go this way we should first verify that okhttp really works when setting the OTLP protocol as http/protobuf instead of the default grpc.

I think you need to have the users control the switch as they also need to reconfigure ports, possibly their tracing backend etc.

Reply:
> I think you need to have the users control the switch as they also need to reconfigure ports, possibly their tracing backend etc.

The users should be already able to do the switch by setting the `OTEL_EXPORTER_OTLP_PROTOCOL` env var (to `http/protobuf`) when starting the bridge. This is valid on bare metal / vm.  With the operator in place, the user could use the `ContainerTemplate` instead to set the env var during the migration phase. When grpc support is dropped we could "hard code" the env var set by the operator itself directly when configuring the `KafkaBridge` instance.

Reply:
> When grpc support is dropped we could "hard code" the env var set by the operator itself directly when configuring the KafkaBridge instance.

I guess defaulting would be nicer than hardcoding it. But yes, it seems like we can quite easily give them time to migrate to HTTP first and then we can drop OkHttp.